## <a id='1'></a>Introduction 

A Vehicle Distributors company has approached us with a dataset analysis task to help them make critical decisions regarding potential future expansion. The objective of this project is to address their inquiries and provide data-driven answers by examining the available data.

The database contains eight tables:

 * Customers: customer data
 * Employees: all employee information
 * Offices: sales office information
 * Orders: customers' sales orders
 * OrderDetails: sales order line for each sales order
 * Payments: customers' payment records
 * Products: a list of scale model cars
 * ProductLines: a list of product line categories

 Table relationships are shown below
![datbase_tables.png](https://github.com/allar14/sales-records-database-analysis-SQL-/blob/main/datbase_tables.png?raw=true)

In [1]:
# Importing all required libraries
import sqlite3
import pandas as pd

In [2]:
!pip install ipython-sql

In [3]:
# loading the extension
%pip install jupysql --quiet
%load_ext sql

# initializing the connection to existing database.
%sql sqlite:///stores.db


[notice] A new release of pip available: 22.3.1 -> 23.2.1
[notice] To update, run: pip3 install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [4]:
# connect to the database
conn = sqlite3.connect('stores.db')

<b>DATA EXPLORATION<b>

In [5]:
# Displaying the first five lines from each table

In [6]:
%%sql
SELECT *
FROM customers LIMIT 5;

Running query in 'sqlite:///stores.db'

+----------------+----------------------------+-----------------+------------------+--------------+------------------------------+--------------+-----------+----------+------------+-----------+------------------------+-------------+
| customerNumber |        customerName        | contactLastName | contactFirstName |    phone     |         addressLine1         | addressLine2 |    city   |  state   | postalCode |  country  | salesRepEmployeeNumber | creditLimit |
+----------------+----------------------------+-----------------+------------------+--------------+------------------------------+--------------+-----------+----------+------------+-----------+------------------------+-------------+
|      103       |     Atelier graphique      |     Schmitt     |     Carine       |  40.32.2555  |        54, rue Royale        |     None     |   Nantes  |   None   |   44000    |   France  |          1370          |    21000    |
|      112       |     Signal Gift Stores     |       King      |       Jean       |  7025551838  |       8489 Strong St.        |     None     | Las Vegas |    NV    |   83030    |    USA    |          1166          |    71800    |
|      114       | Australian Collectors, Co. |     Ferguson    |      Peter       | 03 9520 4555 |      636 St Kilda Road       |   Level 3    | Melbourne | Victoria |    3004    | Australia |          1611          |    117300   |
|      119       |     La Rochelle Gifts      |     Labrune     |     Janine       |  40.67.8555  | 67, rue des Cinquante Otages |     None     |   Nantes  |   None   |   44000    |   France  |          1370          |    118200   |
|      121       |     Baane Mini Imports     |    Bergulfsen   |      Jonas       |  07-98 9555  |    Erling Skakkes gate 78    |     None     |  Stavern  |   None   |    4110    |   Norway  |          1504          |    81700    |
+----------------+----------------------------+-----------------+------------------+--------------+------------------------------+--------------+-----------+----------+------------+-----------+------------------------+-------------+

In [7]:
%%sql
SELECT *
FROM products LIMIT 5;

Running query in 'sqlite:///stores.db'

+-------------+---------------------------------------+--------------+--------------+--------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------+----------+--------+
| productCode |              productName              | productLine  | productScale |      productVendor       |                                                                                                                                                                                                                                                 productDescription                                                                                                                                                                                                                                                | quantityInStock | buyPrice |  MSRP  |
+-------------+---------------------------------------+--------------+--------------+--------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------+----------+--------+
|   S10_1678  | 1969 Harley Davidson Ultimate Chopper | Motorcycles  |     1:10     |     Min Lin Diecast      |                                                                                                                                       This replica features working kickstand, front suspension, gear-shift lever, footbrake lever, drive chain, wheels and steering. All parts are particularly delicate due to their precise scale and require special care and attention.                                                                                                                                      |       7933      |  48.81   |  95.7  |
|   S10_1949  |        1952 Alpine Renault 1300       | Classic Cars |     1:10     | Classic Metal Creations  |                                                                                                                                                                                  Turnable front wheels; steering function; detailed interior; detailed engine; opening hood; opening trunk; opening doors; and detailed chassis.                                                                                                                                                                                  |       7305      |  98.58   | 214.3  |
|   S10_2016  |         1996 Moto Guzzi 1100i         | Motorcycles  |     1:10     | Highway 66 Mini Classics |                                                      Official Moto Guzzi logos and insignias, saddle bags located on side of motorcycle, detailed engine, working steering, working suspension, two leather seats, luggage rack, dual exhaust pipes, small saddle bag located on handle bars, two-tone paint with chrome accents, superior die-cast detail , rotating wheels , working kick stand, diecast metal with plastic parts and baked enamel finish.                                                      |       6625      |  68.99   | 118.94 |
|   S10_4698  |  2003 Harley-Davidson Eagle Drag Bike | Motorcycles  |     1:10     |    Red S

In [8]:
%%sql
SELECT *
FROM productlines LIMIT 5;

Running query in 'sqlite:///stores.db'

+--------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------+-------+
| productLine  |                                                                                                                                                                                                                                                                                                                                                                         textDescription                                                                                                                                                                                                                                                                                                                                                                         | htmlDescription | image |
+--------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------+-------+
| Classic Cars | Attention car enthusiasts: Make your wildest car ownership dreams come true. Whether you are looking for classic muscle cars, dream sports cars or movie-inspired miniatures, you will find great choices in this category. These replicas feature superb attention to detail and craftsmanship and offer features such as working steering system, opening forward compartment, opening rear trunk with removable spare wheel, 4-wheel independent spring suspension, and so on. The models range in size from 1:10 to 1:24 scale and include numerous limited edition and several out-of-production vehicles. All models include a certificate of authenticity from their manufacturers and come fully assembled and ready for display in the home or office. |       None      |  None |
| Motorcycles  |                                                                       Our motorcycles are state of the art replicas of classic as well as contemporary motorcycle legends such as Harley Davidson, Ducati and Vespa. Models contain stunning details such as official logos, rotating wheels, working kickstand, front suspension, gear-shift lever, footbrake lever, and drive chain. Materials used include diecast and plastic. The models range in size from 1:10 to 1:50 scale and include numerous limited edition and several out-of-production vehicles. All models come fully assembled and ready for display in the home or office. Most include a certificate of authenticity.                                                                       |       None      |  None |
|    Planes    |                                                                               

In [9]:
%%sql
SELECT *
FROM orders LIMIT 5;

Running query in 'sqlite:///stores.db'

+-------------+------------+--------------+-------------+---------+------------------------+----------------+
| orderNumber | orderDate  | requiredDate | shippedDate |  status |        comments        | customerNumber |
+-------------+------------+--------------+-------------+---------+------------------------+----------------+
|    10100    | 2003-01-06 |  2003-01-13  |  2003-01-10 | Shipped |          None          |      363       |
|    10101    | 2003-01-09 |  2003-01-18  |  2003-01-11 | Shipped | Check on availability. |      128       |
|    10102    | 2003-01-10 |  2003-01-18  |  2003-01-14 | Shipped |          None          |      181       |
|    10103    | 2003-01-29 |  2003-02-07  |  2003-02-02 | Shipped |          None          |      121       |
|    10104    | 2003-01-31 |  2003-02-09  |  2003-02-01 | Shipped |          None          |      141       |
+-------------+------------+--------------+-------------+---------+------------------------+----------------+

In [10]:
%%sql
SELECT *
FROM orderdetails LIMIT 5;

Running query in 'sqlite:///stores.db'

+-------------+-------------+-----------------+-----------+-----------------+
| orderNumber | productCode | quantityOrdered | priceEach | orderLineNumber |
+-------------+-------------+-----------------+-----------+-----------------+
|    10100    |   S18_1749  |        30       |    136    |        3        |
|    10100    |   S18_2248  |        50       |   55.09   |        2        |
|    10100    |   S18_4409  |        22       |   75.46   |        4        |
|    10100    |   S24_3969  |        49       |   35.29   |        1        |
|    10101    |   S18_2325  |        25       |   108.06  |        4        |
+-------------+-------------+-----------------+-----------+-----------------+

In [11]:
%%sql
SELECT *
FROM payments LIMIT 5;

Running query in 'sqlite:///stores.db'

+----------------+-------------+-------------+----------+
| customerNumber | checkNumber | paymentDate |  amount  |
+----------------+-------------+-------------+----------+
|      103       |   HQ336336  |  2004-10-19 | 6066.78  |
|      103       |   JM555205  |  2003-06-05 | 14571.44 |
|      103       |   OM314933  |  2004-12-18 | 1676.14  |
|      112       |   BO864823  |  2004-12-17 | 14191.12 |
|      112       |   HQ55022   |  2003-06-06 | 32641.98 |
+----------------+-------------+-------------+----------+

In [12]:
%%sql
SELECT *
FROM employees LIMIT 5;

Running query in 'sqlite:///stores.db'

+----------------+-----------+-----------+-----------+---------------------------------+------------+-----------+----------------------+
| employeeNumber |  lastName | firstName | extension |              email              | officeCode | reportsTo |       jobTitle       |
+----------------+-----------+-----------+-----------+---------------------------------+------------+-----------+----------------------+
|      1002      |   Murphy  |   Diane   |   x5800   |   dmurphy@classicmodelcars.com  |     1      |    None   |      President       |
|      1056      | Patterson |    Mary   |   x4611   |  mpatterso@classicmodelcars.com |     1      |    1002   |       VP Sales       |
|      1076      |  Firrelli |    Jeff   |   x9273   |  jfirrelli@classicmodelcars.com |     1      |    1002   |     VP Marketing     |
|      1088      | Patterson |  William  |   x4871   | wpatterson@classicmodelcars.com |     6      |    1056   | Sales Manager (APAC) |
|      1102      |   Bondur  |   Gerard  |   x5408   |   gbondur@classicmodelcars.com  |     4      |    1056   | Sale Manager (EMEA)  |
+----------------+-----------+-----------+-----------+---------------------------------+------------+-----------+----------------------+

In [13]:
%%sql
SELECT *
FROM offices LIMIT 5;

Running query in 'sqlite:///stores.db'

+------------+---------------+-----------------+--------------------------+--------------+------------+---------+------------+-----------+
| officeCode |      city     |      phone      |       addressLine1       | addressLine2 |   state    | country | postalCode | territory |
+------------+---------------+-----------------+--------------------------+--------------+------------+---------+------------+-----------+
|     1      | San Francisco | +1 650 219 4782 |    100 Market Street     |  Suite 300   |     CA     |   USA   |   94080    |     NA    |
|     2      |     Boston    | +1 215 837 0825 |     1550 Court Place     |  Suite 102   |     MA     |   USA   |   02107    |     NA    |
|     3      |      NYC      | +1 212 555 3000 |   523 East 53rd Street   |   apt. 5A    |     NY     |   USA   |   10022    |     NA    |
|     4      |     Paris     | +33 14 723 4404 | 43 Rue Jouffroy D'abbans |     None     |    None    |  France |   75017    |    EMEA   |
|     5      |     Tokyo     | +81 33 224 5000 |       4-1 Kioicho        |     None     | Chiyoda-Ku |  Japan  |  102-8578  |   Japan   |
+------------+---------------+-----------------+--------------------------+--------------+------------+---------+------------+-----------+

<b>DATA ANALYSIS<b>

In [14]:
# Retrieve in a single table the table name as a string, 
# the number of attributes and also the number of rows in each table.

In [15]:
%%sql 
SELECT 'customers' table_name, count(*) number_of_attributes,(SELECT count(*)FROM customers ) number_of_rows FROM pragma_table_info('customers')
UNION ALL
SELECT 'products' table_name, count(*) number_of_attributes,(SELECT count(*)FROM products ) number_of_rows FROM pragma_table_info('products')
UNION ALL
SELECT 'productlines' table_name, count(*) number_of_attributes,(SELECT count(*)FROM productlines ) number_of_rows FROM pragma_table_info('productlines')
UNION ALL
SELECT 'orders' table_name, count(*) number_of_attributes,(SELECT count(*)FROM orders ) number_of_rows FROM pragma_table_info('orders')
UNION ALL
SELECT 'orderdetails' table_name, count(*) number_of_attributes,(SELECT count(*)FROM orderdetails ) number_of_rows FROM pragma_table_info('orderdetails')
UNION ALL
SELECT 'payments' table_name, count(*) number_of_attributes,(SELECT count(*)FROM payments ) number_of_rows FROM pragma_table_info('payments')
UNION ALL
SELECT 'employees' table_name, count(*) number_of_attributes,(SELECT count(*)FROM employees ) number_of_rows FROM pragma_table_info('employees')
UNION ALL
SELECT 'offices' table_name, count(*) number_of_attributes,(SELECT count(*)FROM offices ) number_of_rows FROM pragma_table_info('offices');

Running query in 'sqlite:///stores.db'

+--------------+----------------------+----------------+
|  table_name  | number_of_attributes | number_of_rows |
+--------------+----------------------+----------------+
|  customers   |          13          |      122       |
|   products   |          9           |      110       |
| productlines |          4           |       7        |
|    orders    |          7           |      326       |
| orderdetails |          5           |      2996      |
|   payments   |          4           |      273       |
|  employees   |          8           |       23       |
|   offices    |          9           |       7        |
+--------------+----------------------+----------------+

Question 1: Which products should we order more or less of?

This question refers to inventory reports, including low stock(products in demand) and product performance. 
- The low stock represents the guantity of the sum of each product ordered divided by the
guantitv of product in stock.
- The product performance represents the sum of sales per product.
- Priority products for restocking are those with high product performance that are on the brink of being out ot stock.


In [33]:
%%sql
SELECT p.productCode, p.productName, SUM(od.quantityOrdered)/p.quantityInStock as low_stock , ROUND(SUM(od.quantityOrdered*priceEach),2) as product_performance
FROM products p
JOIN orderdetails od
ON p.productCode = od.productCode
GROUP BY p.productCode
ORDER BY product_performance DESC;

Running query in 'sqlite:///stores.db'

+-------------+--------------------------------------+-----------+---------------------+
| productCode |             productName              | low_stock | product_performance |
+-------------+--------------------------------------+-----------+---------------------+
|   S18_3232  |     1992 Ferrari 360 Spider red      |     0     |      276839.98      |
|   S12_1108  |          2001 Ferrari Enzo           |     0     |      190755.86      |
|   S10_1949  |       1952 Alpine Renault 1300       |     0     |      190017.96      |
|   S10_4698  | 2003 Harley-Davidson Eagle Drag Bike |     0     |       170686.0      |
|   S12_1099  |          1968 Ford Mustang           |     13    |      161531.48      |
|   S12_3891  |           1969 Ford Falcon           |     0     |      152543.02      |
|   S18_1662  |     1980s Black Hawk Helicopter      |     0     |      144959.91      |
|   S18_2238  |    1998 Chrysler Plymouth Prowler    |     0     |      142530.63      |
|   S18_1749  |       1917 Grand Touring Sedan       |     0     |       140535.6      |
|   S12_2823  |           2002 Suzuki XREO           |     0     |      135767.03      |
+-------------+--------------------------------------+-----------+---------------------+
Truncated to displaylimit of 10.

In [34]:
%%sql
SELECT * 
FROM (SELECT p.productCode, p.productName, ROUND(SUM(od.quantityOrdered)/p.quantityInStock, 2) as low_stock
      FROM products p
      JOIN orderdetails od
      ON p.productCode = od.productCode
      GROUP BY p.productCode
      ORDER BY low_stock DESC);

Running query in 'sqlite:///stores.db'

+-------------+---------------------------------------------+-----------+
| productCode |                 productName                 | low_stock |
+-------------+---------------------------------------------+-----------+
|   S24_2000  |           1960 BSA Gold Star DBD34          |    67.0   |
|   S12_1099  |              1968 Ford Mustang              |    13.0   |
|   S32_4289  |           1928 Ford Phaeton Deluxe          |    7.0    |
|   S32_1374  |               1997 BMW F650 ST              |    5.0    |
|   S72_3212  |                  Pont Yacht                 |    2.0    |
|  S700_3167  |              F/A 18 Hornet 1/72             |    1.0    |
|  S700_1938  |                The Mayflower                |    1.0    |
|   S50_4713  |              2002 Yamaha YZR M1             |    1.0    |
|   S32_3522  | 1996 Peterbilt 379 Stake Bed with Outrigger |    1.0    |
|   S18_2795  |            1928 Mercedes-Benz SSK           |    1.0    |
+-------------+---------------------------------------------+-----------+
Truncated to displaylimit of 10.

In [35]:
%%sql
SELECT * 
FROM (SELECT productCode, ROUND(SUM(quantityOrdered*priceEach), 2) as product_performance
      FROM orderdetails
      GROUP BY productCode
      ORDER BY product_performance DESC);

Running query in 'sqlite:///stores.db'

+-------------+---------------------+
| productCode | product_performance |
+-------------+---------------------+
|   S18_3232  |      276839.98      |
|   S12_1108  |      190755.86      |
|   S10_1949  |      190017.96      |
|   S10_4698  |       170686.0      |
|   S12_1099  |      161531.48      |
|   S12_3891  |      152543.02      |
|   S18_1662  |      144959.91      |
|   S18_2238  |      142530.63      |
|   S18_1749  |       140535.6      |
|   S12_2823  |      135767.03      |
+-------------+---------------------+
Truncated to displaylimit of 10.

In [41]:
%%sql
WITH lowstock_table AS (SELECT p.productCode
FROM (SELECT p.productCode, p.productName, ROUND(SUM(od.quantityOrdered)/p.quantityInStock, 2) as low_stock
      FROM products p
      JOIN orderdetails od
      ON p.productCode = od.productCode
      GROUP BY p.productCode
      ORDER BY low_stock DESC)
),

PP_table AS (SELECT productCode
FROM (SELECT productCode, ROUND(SUM(quantityOrdered*priceEach), 2) as product_performance
      FROM orderdetails
      GROUP BY productCode
      ORDER BY product_performance DESC)
)

SELECT p.productCode, p.productName, p.productLine
FROM products p
WHERE p.productCode in lowstock_table
AND p.productCode in PP_table;

Running query in 'sqlite:///stores.db'

+-------------+---------------------------------------+------------------+
| productCode |              productName              |   productLine    |
+-------------+---------------------------------------+------------------+
|   S10_1678  | 1969 Harley Davidson Ultimate Chopper |   Motorcycles    |
|   S10_1949  |        1952 Alpine Renault 1300       |   Classic Cars   |
|   S10_2016  |         1996 Moto Guzzi 1100i         |   Motorcycles    |
|   S10_4698  |  2003 Harley-Davidson Eagle Drag Bike |   Motorcycles    |
|   S10_4757  |          1972 Alfa Romeo GTA          |   Classic Cars   |
|   S10_4962  |         1962 LanciaA Delta 16V        |   Classic Cars   |
|   S12_1099  |           1968 Ford Mustang           |   Classic Cars   |
|   S12_1108  |           2001 Ferrari Enzo           |   Classic Cars   |
|   S12_1666  |             1958 Setra Bus            | Trucks and Buses |
|   S12_2823  |            2002 Suzuki XREO           |   Motorcycles    |
+-------------+---------------------------------------+------------------+
Truncated to displaylimit of 10.

Question 2: How Should we Match marketing Communication Strategies to Customer Behaviour

in the first part of this project. we explored products. Now we'll explore customer intormation by answering the second question: how should we match marketing and communication strategies
to customer behaviors? This involves categorizing customers: finding the VIP (very important person) customers and those who are less engaged.

- VIP customers bring in the most profit for the store.
- Less-engaged customers bring in less profit.

Lets compute how much profit each customer generates using the following tables: 'Products', 'Orderdetails', 'Orders'.

In [42]:
%%sql
SELECT o.customerNumber, SUM(od.quantityOrdered * (od.priceEach - p.buyPrice)) as profit
FROM products p
JOIN orderdetails od 
ON p.productCode = od.productCode
JOIN orders o 
ON od.orderNumber = o.orderNumber
GROUP BY o.customerNumber
ORDER BY profit DESC;

Running query in 'sqlite:///stores.db'

+----------------+--------------------+
| customerNumber |       profit       |
+----------------+--------------------+
|      141       | 326519.65999999986 |
|      124       | 236769.38999999998 |
|      151       | 72370.09000000001  |
|      114       | 70311.06999999999  |
|      119       | 60875.30000000001  |
|      148       | 60477.37999999999  |
|      187       | 60095.85999999999  |
|      323       |      60013.99      |
|      131       | 58669.10000000001  |
|      450       | 55931.369999999995 |
+----------------+--------------------+
Truncated to displaylimit of 10.

VIP CUSTOMERS

In [43]:
%%sql


WITH profit_table as (SELECT o.customerNumber, SUM(od.quantityOrdered * (od.priceEach - p.buyPrice)) as profit
                      FROM products p
                      JOIN orderdetails od 
                      ON p.productCode = od.productCode
                      JOIN orders o 
                      ON od.orderNumber = o.orderNumber
                      GROUP BY o.customerNumber
                      ORDER BY profit DESC
                      LIMIT 5
)

SELECT c.contactLastName, c.contactFirstName, c.city, c.country, pt.profit
FROM customers c
JOIN profit_table pt
ON c.customerNumber = pt.customerNumber
;



Running query in 'sqlite:///stores.db'

+-----------------+------------------+------------+-----------+--------------------+
| contactLastName | contactFirstName |    city    |  country  |       profit       |
+-----------------+------------------+------------+-----------+--------------------+
|      Freyre     |      Diego       |   Madrid   |   Spain   | 326519.65999999986 |
|      Nelson     |      Susan       | San Rafael |    USA    | 236769.38999999998 |
|      Young      |       Jeff       |    NYC     |    USA    | 72370.09000000001  |
|     Ferguson    |      Peter       | Melbourne  | Australia | 70311.06999999999  |
|     Labrune     |     Janine       |   Nantes   |   France  | 60875.30000000001  |
+-----------------+------------------+------------+-----------+--------------------+

LESS ENGAGED CUSTOMERS

In [44]:
%%sql


WITH profit_table as (SELECT o.customerNumber, SUM(od.quantityOrdered * (od.priceEach - p.buyPrice)) as profit
                      FROM products p
                      JOIN orderdetails od 
                      ON p.productCode = od.productCode
                      JOIN orders o 
                      ON od.orderNumber = o.orderNumber
                      GROUP BY o.customerNumber
                      ORDER BY profit 
                      LIMIT 5
)

SELECT c.contactLastName, c.contactFirstName, c.city, c.country, pt.profit
FROM customers c
JOIN profit_table pt
ON c.customerNumber = pt.customerNumber;

Running query in 'sqlite:///stores.db'

+-----------------+------------------+------------+---------+--------------------+
| contactLastName | contactFirstName |    city    | country |       profit       |
+-----------------+------------------+------------+---------+--------------------+
|      Young      |       Mary       |  Glendale  |   USA   | 2610.870000000001  |
|      Taylor     |      Leslie      | Brickhaven |   USA   | 6586.0199999999995 |
|     Ricotti     |      Franco      |   Milan    |  Italy  |      9532.93       |
|     Schmitt     |     Carine       |   Nantes   |  France | 10063.800000000001 |
|      Smith      |     Thomas       |   London   |    UK   | 10868.039999999999 |
+-----------------+------------------+------------+---------+--------------------+

Question 2: How much can we spend on acquiring new customers?

To determine how much money we can spend acquiring new customers, we can compute the customer Lifetime value LTV which represents the average amount of money a customer generates. 
We can then determine how much we can spend on marketing.

In [45]:
%%sql


WITH profit_table as (SELECT o.customerNumber, SUM(od.quantityOrdered * (od.priceEach - p.buyPrice)) as profit
                      FROM products p
                      JOIN orderdetails od 
                      ON p.productCode = od.productCode
                      JOIN orders o 
                      ON od.orderNumber = o.orderNumber
                      GROUP BY o.customerNumber
                      ORDER BY profit 
)

SELECT AVG(profit)
FROM profit_table;



Running query in 'sqlite:///stores.db'

+------------------+
|   AVG(profit)    |
+------------------+
| 39039.5943877551 |
+------------------+

This LTV is an average profit generated by each customer over the time they have been with the business which gives us and idea and help us make predictions like if we get 50 new customers, can we guarantee (50*39039.59) amount of additional profit.

<h>CONCLUSION<h>


PRODUCTS
- We see that Classic cars are dominant in the results generated. Overall, the demand for classic cars is driven by a combination of emotional attachment, investment potential, and a deep appreciation for their historical and aesthetic qualities. These factors contribute to the enduring popularity of classic cars among collectors and enthusiasts making it a highly demanded product which the business should pay attention to and ensure there is a generous quantity available instock at all times.
- For the products that are not considered low stock and high performing requires more indepth analysis to determing why customers are not buying these products so as to decide if some internal changes are needed or curated offers are necessary to boost sales or if the business needs to let go of these productlines.



CUSTOMERS

- VIP : It's important to recognize and appreciate the loyalty of customers to a business as this impacts the business positively in the long run. For the VIP customers, organizing exclusive events as well as rewarding them with special offers to boost their perceived feeling of importance and guarantee continued stay with the business is a good way to recognize this group of customers. 

- LESS ENGAGING: For the less engaging customers, curation discounted offer and time conscious bonuses can encourage them to be more active and potentially strengthen their relationship with the business. Understanding what this group of customers value more and their need would enable us curate more tailored offers as well so a little bit of research into the customer wants is highly advisable.